### ISSUE TITLE
Can't change subscribers list

### ISSUE THREAD

**Issue by kolewu** [mziech/nextcloud-majordomo#22]
I can't use the application to change the subscribers list by adding email addresses (can't use nextcloud users or groups, as this is just the test instance without users and even in production I don't want to add all mailinglist subscribers as users to my nextcloud). Maybe I don't understand the workflow, but even some peek into the code hasn't changed that.

- there is
  - a working IMAP configuration
  - a working mailinglist definition
- If I "retrieve current status" I get the already subscribed address as "Additional E-Mail" under "Member Policy" and "Current list members"

But how to proceed from here? If I add another email by adding a new "Additional E-Mail" policy, "Write desired changes" gives me "There are currently no pending changes."

Yesterday I have one time seen a list of actions, but since there was an error, I have canceled it. The error (if I remember it correctly) was caused by me, when first adding a new address and then deleting it and adding it again, the displayed action was "Additional E-Mail" + "Exclude E-Mail" for this address ...

I have even inspected the database, but there is only the current member (added without the app to test the import) for both the _members_ and the _who_ table and nothing changed when adding new policies.

**Comment by mziech** [comment]
Well, the original idea of the app is to sync the Nextcloud user database to a Majordomo instance, so you define the policies once and then use the Nextcloud user management to transparently (un)subscribe the users to the mailing lists by (un)assigning them to groups.

If you don't want to give everybody on the mailing list a Nextcloud account, that's possible - as you noted - using "additional" email addresses. This is also what the import will create since it will not be able to assign the existing email addresses to any existing Nextcloud user.

Basically, your workflow seems to be correct. Have you tried hitting "Save" before clicking the button to write the changes and does that fix your problem?

**Comment by kolewu** [comment]
Clicking "Save" was the missing piece. It's not obvious, that "Save" works on the policies, too. For me it's only related to the settings above. And maybe this was also part of the confusion before, because I tested different settings and pressed "Save" with some artifacts in the policy-list.
So after adding a new policy (in my case a new email address) when I click "Save" then the policy will be added to the database and the "Write desired changes" show the newly added address and subscribe it to the list. Very nice!

**Comment by mziech** [comment]
I agree, despite completely changing the UX it may be easiest to lock all buttons except "Save" when a policy has changed ...
